### PR TITLE
[readtapi] Add Extract & Remove architecture functionality

### PR DESCRIPTION
**Pull request by cyndyishida**
NOTE: This is a part of a stack. Please review https://github.com/llvm/llvm-project/pull/72656 first.

This adds functionality to tbd files similar to what `lipo -extract`
and `lipo -remove` does for binaries.

### CODE REVIEW

File: llvm/tools/llvm-readtapi/llvm-readtapi.cpp [hunk 1]
```diff
@@ -151,8 +171,10 @@ int main(int Argc, char **Argv) {
         exit(1);
       });
   if (Args.hasArg(OPT_help)) {
-    Tbl.printHelp(outs(), "llvm-readtapi [options] <inputs>",
-                  "LLVM TAPI file reader and manipulator");
+    Tbl.printHelp(
+        outs(),
+        "llvm-readtapi [options]* [-arch <arch>]* <inputs> [-o <output>]*",
```
Comment (ributzka) [inline llvm/tools/llvm-readtapi/llvm-readtapi.cpp:176]: Isn't the `*` implied when `[options]` is used? There can be multiple outputs?

Comment (cyndyishida) [reply llvm/tools/llvm-readtapi/llvm-readtapi.cpp]: I guess I used * to denote 0 or more flags, in the case for output, no -o means write to stdout. But yes you're right, I don't need one for [options].

**Review by ributzka — APPROVED** [review]
LGTM. Could you please add a comment at the beginning of each test and summarize what is being tested?

**Comment by cyndyishida** [comment]
> LGTM. Could you please add a comment at the beginning of each test and summarize what is being tested?

Addressed in https://github.com/llvm/llvm-project/pull/73576

**Pull request merged**